openhands/core/config.py

```diff
@@ -70,7 +70,7 @@ class LLMConfig:
     retry_min_wait: int = 15
     retry_max_wait: int = 120
     timeout: int | None = None
-    max_message_chars: int = 10_000  # maximum number of characters in an observation's content when sent to the llm
+    max_message_chars: int = 20_000  # maximum number of characters in an observation's content when sent to the llm
```

Comment:
Please note that we have discussed this 10_000 in the past: https://github.com/All-Hands-AI/OpenHands/issues/2540

Reply:
I understand this comment, but if our accuracy goes up then I'm inclined to just go ahead and accept this PR and then we could always do an ablation study later reducing to 10k

Reply:
I think this refers to the number of characters in each "OBSERVATION"? The reason for this was bc sometimes `pip install` will give an extremely long history & boomed the context window. I'd suggest maybe we could start a separate PR and try to experiment a much higher limits (e.g., 100k)?

Reply:
yeah - if we get this bundle to work better than main i think it is probably good to accept this now

Reply:
I don't want to make this more complicated as it is, I'll revert it back to 10K and we can track it in extra PR for followup, but this discussion should be had (with eval, too 😂 ).

Reply:
> I think this refers to the number of characters in each "OBSERVATION"? The reason for this was bc sometimes `pip install` will give an extremely long history & boomed the context window. I'd suggest maybe we could start a separate PR and try to experiment a much higher limits (e.g., 100k) by running eval against it?

The observation is the way for `open_file` to show the content to the LLM, so _IF_ the LLM chooses to retrieve more lines in a single take (seldomly happens), this limit could be reached, just as a sidenote.